provider/ec2/internal/testing/iam_server.go

```diff
@@ -29,6 +30,12 @@ func NewIAMServer() (*IAMServer, error) {
 	return srv, nil
 }
 
+func (i *IAMServer) ProducePermissionError(p bool) {
+	i.mu.Lock()
+	defer i.mu.Unlock()
+	i.producePermissionError = p
+}
+
```

Comment:
Yuck :-)
Can we instead have it return a permission error based on something like the prefix being used? Or the id value or something we can set as part of the role or profile?

Reply:
Not really